src/infra/flow-system/inmem/src/flow_configuration/flow_configuration_event_store_inmem.rs

```diff
@@ -94,7 +94,16 @@ impl FlowConfigurationEventStore for FlowConfigurationEventStoreInMem {
     async fn list_all_dataset_ids(&self) -> FailableDatasetIDStream {
         use futures::StreamExt;
 
-        let dataset_ids = self.inner.as_state().lock().unwrap().dataset_ids.clone();
+        let dataset_ids: Vec<_> = self
+            .inner
+            .as_state()
+            .lock()
+            .unwrap()
+            .dataset_ids
+            .iter()
+            .rev()
```

Comment:
From my understanding, this function returns a "set", not  a "sequence", so there is no meaningful order.
I would suggest sorting in tests directly.